### DISCUSSION TITLE
padding-line-between-statements for shorthand methods

### DISCUSSION THREAD

**Discussion by dmke** [eslint/eslint#15533] [Q&A] [answered]
I'm trying to get eslint to format this object literal to have an empty line before and between the functions, but not between ordinary properties:

<details><summary>Incorrect example</summary>

```js
const foo = {
  a: 42,
  b: "a string",
  bar() {
    // ...
  },
  baz() {
    // ...
  },
}
```

</details>
<details open><summary>Correct example</summary>

```js
const foo = {
  a: 42,
  b: "a string",

  bar() {
    // ...
  },

  baz() {
    // ...
  },
}
```

</details>

I currently have these rules:

```json
{
  "rules": {
    "padding-line-between-statements": ["error",
      { "blankLine": "always", "prev": "block-like", "next": "*" },
      { "blankLine": "never", "prev": "block-like", "next": ["block-like", "break", "return"] },
      { "blankLine": "always", "prev": ["break", "return"], "next": "case" },
      { "blankLine": "always", "prev": "function", "next": "function" }
    ],
  }
}
```

Is that possible? Looking at the [`STATEMENT_TYPE`](https://eslint.org/docs/rules/padding-line-between-statements#rule-details) list for that rule, I don't see a "prop", "object-method", "shorthand-function" or similar names. "function" seems to apply only to ordinary `function() {}` declarations.

**Comment by aladdin-add** [comment, marked as answer]
it's not in the scope of this rule as `a: 42,` is not a statement. you'll need to make a new rule to enforce the style.

**Reply by dmke** [reply]
Hm, then technically the method definitions should be out of scope as well, since these are equivalent:

```js
const shorthand = {
  bar() {}
}

const expanded = {
  bar: function() {}
}
```

What would be the correct rule for this, if there is one? I haven't yet looked through all rules, but some obvious ones standing out are:

- [`object-shorthand`](https://eslint.org/docs/rules/object-shorthand) seems to be concerned with transforming between shorthand and expanded (normal?) form
- [`padded-blocks`](https://eslint.org/docs/rules/padded-blocks) doesn't apply to object literals
- [`object-curly-spacing`](https://eslint.org/docs/rules/object-curly-spacing) and [`object-curly-newline`](https://eslint.org/docs/rules/object-curly-newline) add or remove white spaces before/after closing/opening curlies
- [`no-whitespace-before-property`](https://eslint.org/docs/rules/no-whitespace-before-property) is used for chained calls (`foo.bar.baz` vs. `foo .bar .baz`)
- [`object-property-newline`](https://eslint.org/docs/rules/object-property-newline) comes close, but only enforces properties starting on a new line

**Reply by aladdin-add** [reply]
I think there is no core rule to enforce it, and stylistic rules are frozen per our [rules policies](https://eslint.org/blog/2020/05/changes-to-rules-policies).

**Reply by dmke** [reply]
I see, thank you.
